src/service.js

```diff
@@ -79,19 +79,17 @@ export function getServicePromiseOrNull(win, id) {
  * Returns a service for the given id and ampdoc (a per-ampdoc singleton).
  * If the service is not yet available the factory function is invoked and
  * expected to return the service.
- * @param {!Node} node
+ * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrDoc
  * @param {string} id of the service.
  * @param {function(!AmpDoc):!Object=} opt_factory Should create the service
  *     if it does not exist yet. If the factory is not given, it is an error
  *     if the service does not exist yet.
  * @return {*}
  */
-export function getServiceForDoc(node, id, opt_factory) {
-  const win = node.ownerDocument.defaultView;
-  const ampdocService = getAmpdocService(win);
-  const ampdoc = ampdocService.getAmpDoc(node);
+export function getServiceForDoc(nodeOrDoc, id, opt_factory) {
```

Comment:
basically make it so that if you have an AmpDoc, you can always get a
service. If the service is window scoped, it just gets it off the window.

In the other comment we decided not to do that, though :)

On Fri, Jun 24, 2016 at 9:01 AM, Dima Voytenko <notifications@github.com>
wrote:

> In src/service.js
> <https://github.com/ampproject/amphtml/pull/3769#discussion_r68420347>:
>
> >   * @param {string} id of the service.
> >   * @param {function(!AmpDoc):!Object=} opt_factory Should create the service
> >   *     if it does not exist yet. If the factory is not given, it is an error
> >   *     if the service does not exist yet.
> >   * @return {*}
> >   */
> > -export function getServiceForDoc(node, id, opt_factory) {
> > -  const win = node.ownerDocument.defaultView;
> > -  const ampdocService = getAmpdocService(win);
> > -  const ampdoc = ampdocService.getAmpDoc(node);
> > +export function getServiceForDoc(nodeOrDoc, id, opt_factory) {
>
> Sorry, missed that note. Allow what for getService to pass the ampdoc?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/3769/files/4340a79da0979e325280e67eeb0be999857443e5#r68420347>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAFeT9PpX9FcTm1cv6UOnQ63LicNFJBsks5qO_9agaJpZM4I9cTo>
> .
>